src/debug/debugSession.ts

```diff
@@ -104,6 +105,97 @@ export class DebugSession implements IDebugSession {
         });
     }
 
+    /**
+     * In attach mode, the user should choose the running pod first, then the debugger will attach to it.
+     *
+     * @param workspaceFolder the active workspace folder.
+     * @param pod the target pod name.
+     */
+    public async attach(workspaceFolder: vscode.WorkspaceFolder, pod?: string): Promise<void> {
+        if (!workspaceFolder) {
+            return;
+        }
+
+        // Select the image type to attach.
+        this.debugProvider = await providerRegistry.getDebugProvider();
+        if (!this.debugProvider) {
+            return;
+        } else if (!await this.debugProvider.isDebuggerInstalled()) {
+            return;
+        }
+
+        // Select the target pod to attach.
+        let targetPod, targetContainer, containers = [];
+        if (pod) {
+            targetPod = pod;
+            const shellResult = await this.kubectl.invokeAsync(`get pod/${pod} -o json`);
+            if (shellResult.code !== 0) {

```

Comment:
DUP code for this, you can make a private util function 
+    const shellResult = await this.kubectl.invokeAsync("get pods -o json");
+            if (shellResult.code !== 0) {
+                vscode.window.showErrorMessage(shellResult.stderr);
+                return;
+            }
+            const podObj = JSON.parse(shellResult.stdout.trim());